drivers/charger/bq24190.c

```diff
@@ -0,0 +1,491 @@
+	}
+
+	v = FIELD_GET(BQ24190_REG_PCTCC_ITERM_MASK, v);
+
+	*tc = (v * BQ24190_REG_PCTCC_ITERM_STEP_UA) + BQ24190_REG_PCTCC_ITERM_OFFSET_UA;
+
+	return 0;
+}
+
+static int bq24190_get_cvc(const struct device *dev, uint32_t *voltage)
+{
+	const struct bq24190_config *const config = dev->config;
+	uint8_t v;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, BQ24190_REG_CVC, &v);
+	if (ret < 0) {
+		return ret;
+	}
+
+	v = FIELD_GET(BQ24190_REG_CVC_VREG_MASK, v);
+
+	*voltage = (v * BQ24190_REG_CVC_VREG_STEP_UV) + BQ24190_REG_CVC_VREG_OFFSET_UV;
+
+	return 0;
+}
+
+static int bq24190_set_ccc(const struct device *dev, uint32_t curr)
+{
+	const struct bq24190_config *const config = dev->config;
+	uint8_t v;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, BQ24190_REG_CCC, &v);
+	if (ret < 0) {
+		return ret;
+	}
+
+	v &= BQ24190_REG_CCC_FORCE_20PCT_MASK;
+
+	if (v) {
+		curr *= 5;
+	}
+
+	curr = CLAMP(curr, BQ24190_REG_CCC_ICHG_MIN_UA, BQ24190_REG_CCC_ICHG_MAX_UA);
+
+	v = (curr - BQ24190_REG_CCC_ICHG_OFFSET_UA) / BQ24190_REG_CCC_ICHG_STEP_UA;
+
+	v = FIELD_PREP(BQ24190_REG_CCC_ICHG_MASK, v);
+
+	return i2c_reg_update_byte_dt(&config->i2c, BQ24190_REG_CCC, BQ24190_REG_CCC_ICHG_MASK, v);
+}
+
+static int bq24190_set_cvc(const struct device *dev, uint32_t voltage)
+{
+	const struct bq24190_config *const config = dev->config;
+	uint8_t v;
+
+	voltage = CLAMP(voltage, BQ24190_REG_CVC_VREG_MIN_UV, BQ24190_REG_CVC_VREG_MAX_UV);
+
+	v = (voltage - BQ24190_REG_CVC_VREG_OFFSET_UV) / BQ24190_REG_CVC_VREG_STEP_UV;
+
+	v = FIELD_PREP(BQ24190_REG_CVC_VREG_MASK, v);
+
+	return i2c_reg_update_byte_dt(&config->i2c, BQ24190_REG_CVC, BQ24190_REG_CVC_VREG_MASK, v);
+}
+
+static int bq24190_set_config(const struct device *dev)
+{
+	struct bq24190_data *data = dev->data;
+	union charger_propval val;
+	int ret;
+
+	val.const_charge_curr_ua = data->ichg_ua;
+
+	ret = bq24190_set_ccc(dev, val.const_charge_curr_ua);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val.const_charge_volt_uv = data->vreg_uv;
+
+	return bq24190_set_cvc(dev, val.const_charge_volt_uv);
+}
+
+static int bq24190_get_prop(const struct device *dev, charger_prop_t prop,
+			    union charger_propval *val)
+{
+	switch (prop) {
+	case CHARGER_PROP_ONLINE:
+		return bq24190_charger_get_online(dev, &val->online);
+	case CHARGER_PROP_CHARGE_TYPE:
+		return bq24190_charger_get_charge_type(dev, &val->charge_type);
+	case CHARGER_PROP_HEALTH:
+		return bq24190_charger_get_health(dev, &val->health);
+	case CHARGER_PROP_STATUS:
+		return bq24190_charger_get_status(dev, &val->status);
+	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT:
+		return bq24190_charger_get_ccc(dev, &val->const_charge_curr_ua);
+	case CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE:
+		return bq24190_get_cvc(dev, &val->const_charge_volt_uv);
+	case CHARGER_PROP_PRECHARGE_CURRENT:
+		return bq24190_charger_get_pc(dev, &val->precharge_curr_ua);
+	case CHARGER_PROP_CHARGE_TERM_CURRENT:
+		return bq24190_charger_get_tc(dev, &val->charge_term_curr_ua);
```

Comment:
the first 4 have a full name (`get_charge_type`...) and these are all shortened (`get_pc`, `get_tc`...). Use the expanded name for everything for coherency?

Reply:
ack